src/app/order/order-detail/order-detail-card/order-detail-card.component.ts

```diff
@@ -45,11 +45,15 @@ export class OrderDetailCardComponent implements OnInit {
 			this.order = await this._orderService.update(this.order.id, {
 				branch: newBranchId,
 			});
-			const customerItems = await this._customerItemService.getManyByIds(
-				this.order.orderItems.map((orderItem) =>
-					String(orderItem.customerItem)
+			const customerItems = (
+				await Promise.all(
+					this.order.orderItems.map((orderItem) =>
+						this._customerItemService.get({
+							query: `?blid=${orderItem.blid}&returned=false&buyout=false&cancel=false`,
```

Comment:
To me, it does not seem sufficient to check only blid, when 74k out of 110k orders have orderItems without blid, and some were updated as recently as yesterday. What was wrong with the ID? Could the query be a union of both or something?

Reply:
Sooo, our database is kinda whack: It is only when we have an actual handout order that we include the blid. That is, when the orderItem.type is "partly-payment" or "rent". And that is the only order type that you actually want to change. (since we do not really care about updating the branch for completed orders. eg. orders that are returned or bought out)

Reply:
Therefore, it is kind of redundant to include it, other than for completeness I guess. But there is afaik as I know no business value in updating the branch on completed orders. And if there is, it is sufficient to only update the visible value on the order, namely the branch on the "order" object.